### PR TITLE
Add process metadata file for work with eBPF agent

### PR DESCRIPTION
**Pull request by mrproliu**
Now we have [skywalking-rover](https://github.com/apache/skywalking-rover), which could detect processes from the local machine. Also, go2sky is one of the official agents in the skywalking ecosystem. We could let these two projects cooperate. If the local machine already has service use go2sky, then the rover could find this process automatically. Finally, the rover side could collect, profiling, etc. 

In principle, when go2sky performs keep alive with the backend, it would write a metadata file to the local (temporary directory) at the same time, which describes the information of the current process. The Rover side scans all processes, find out which processes contain this metadata file, and update the confirmed files. Finally, if go2sky sees that it has been confirmed, it would not modify the files again.

### CODE REVIEW

File: reporter/process.go
Comment (wu-sheng) [inline reporter/process.go:70]: I think we need a very clear option for this new feature. Labels are optional, this should not

Comment (wu-sheng) [reply reporter/process.go:70]: And please document clearly what file(s) is being created, and which information would be written.
Then when the rover side doc is ready, this should link to there for further information.

Comment (mrproliu) [reply reporter/process.go]: > I think we need a very clear option for this new feature. Labels are optional, this should not

Yes, this function is not optional, maybe the function name let you feel confused. updated.

Comment (mrproliu) [reply reporter/process.go]: > And please document clearly what file(s) is being created, and which information would be written. Then when the rover side doc is ready, this should link to there for further information.

Updated. After this merge, I will update the documentation on the rover side and finally link the document there.

File: README.md
Comment (wu-sheng) [inline README.md:287]: I think don't need to be so complex, how about putting labels in the file directly?

File: README.md
Comment (wu-sheng) [inline README.md:290]:
```suggestion
### ACK file
```

Comment (wu-sheng) [reply README.md:290]: Please change file and description accordingly. Confirmation feels strange.

File: README.md
Comment (wu-sheng) [inline README.md:292]:
```suggestion
The confirmation file is to let the go2sky know, the rover side has detected the current process, so don't need to keep updating the metadata file again. It save in: `{TMPDIR}/apache_skywalking/process/{pid}/metadata-confirm.properties`.
```

Same issue here.

Also, I am not sure this file `doesn't need to keep udpate`? Then what happens if rover crashes or exits?

File: README.md
Comment (wu-sheng) [inline README.md:267]:
```suggestion
## Process Status Hook
```

Process is a common and wide used concept in Linux, I don't want to hijack it as a feature name.

File: README.md
Comment (wu-sheng) [inline README.md:269]: I think you missed one, I suggest this feature(temp files) should be optional and OFF on default.

File: README.md
Comment (mrproliu) [reply README.md:292]: > Then what happens if rover crashes or exits?

Yes, that's a problem. In this way, maybe we don't need to have the ack mechanism. because it useless to go2sky? WDYT?

Comment (wu-sheng) [reply README.md:292]: Agree, that increases the complexity.

File: README.md
Comment (arugal) [reply README.md:269]: It is recommended to enable this feature through environment variable

File: README.md
Comment (mrproliu) [reply README.md]: Deleted the ack mechanism.

File: README.md
Comment (mrproliu) [reply README.md]: Update to the as the `labels` property key, and write the label values directly.

File: README.md
Comment (mrproliu) [reply README.md]: Updated.

File: README.md
Comment (mrproliu) [reply README.md:271]: Updated and let the process label and process status hook enabled could update through the environment variable.

File: README.md
Comment (mrproliu) [reply README.md]: Only the doc typo, updated.

**Comment by wu-sheng** [comment]
I think 1.4.1 is released, this PR is in the wrong milestone. 

@arugal Do we need plugin 1.4.1 release?

File: README.md
Comment (wu-sheng) [inline README.md:274]:
```suggestion
When go2sky keeps alive with the backend, it would write a metadata file to the local (temporary directory) at the same time, which provides the information of the current process.
The rover would detect this file in order to identify this Golang service, and tags it `ebpf-profiling-able`.
```

As a doc, we don't need to describe how codes work, they are open sourced :) We need to tell why it works in this way.

File: reporter/grpc_env.go
Comment (wu-sheng) [inline reporter/grpc_env.go:37]: What happens if this env variable is set, but no `WithProcessLabels` called in the codes?

File: README.md
Comment (mrproliu) [reply README.md:274]: `ebpf-profiling-able`? Do you mean to add this tag to the process labels?

File: reporter/grpc_env.go
Comment (mrproliu) [reply reporter/grpc_env.go:37]: It could be detected and triggered `WithProcessLabels`. please take a look on: https://github.com/SkyAPM/go2sky/pull/156/files#diff-758d4819d25e782438ebef78193df4c798a5a0b247693e5f44ae9a514e353d53R82-R93

File: README.md
Comment (wu-sheng) [reply README.md:274]: I mean logically, this is what this is designed for. Nothing about label.

**Review by wu-sheng — APPROVED** [review]
Waiting for @arugal 's approval.

**Comment by arugal** [comment]
> @arugal Do we need plugin 1.4.1 relea se?

Yes, I left out

**Review by arugal — APPROVED** [review]

**Pull request merged**
